### PR TITLE
Add tests for PreImage (continued)

### LINKED ISSUE #453: Improve preimage functionality test coverage
As part of the governance E2E test suite introduced in #115, some tests to preimage functionality were added:

https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/d35064e25ba5c162880253858bd0d5d906ae63fc/packages/shared/src/governance.ts#L952-L960

For this issue, the goal is more preimage test coverage.

Examples:
1. preimage requesting/unrequesting
    - in particular, the source for [`unrequest_preimage`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/preimage/src/lib.rs#L253) says it must **not** be called more times than `request_preimage`. Interesting edge case to cover
2. more noting/unnoting preimage tests
    - in particular, edge cases: empty preimages, repeated preimages, preimages with the largest allowed size, and beyond the largest size
    - noting a preimage twice, unnoting one twice, unnoting nonexistent hash
3. testing preimage state transitions triggered by requesting/unnoting of an existing preimage
    - if a preimage is noted without having been requested, it will demand a deposit from the caller, which will be held until the preimage is requested, or unnoted
    - `ensure_updated` iterates over all preimages and updates their states according to whether they've been requested since having been submitted
        - if over 90% of preimages are updated, the caller pays no fee - it would be good to test this, too

### PR DESCRIPTION
**Pull request by andreitrand**
Add additional tests which verify PreImage behavior:
- 1) A manager account requests a preimage hash once then unrequests it multiple times (added additional checks)
- 2) Check the noting and unnoting of empty images
- 3) Check the noting and unnoting of oversized images
- 4) Check the repeated noting and unnoting of the same preimage
- 5) Check preimage behavior when requesting, noting, unrequesting and unnoting occur in different orders
- 6) Check the "ensure_updated" extrinsic applies fees depending on the update ratio

Several utility functions have been moved to the helpers file as they are shared with other test files.

This is a continuation of the PR: https://github.com/open-web3-stack/polkadot-ecosystem-tests/pull/470.
Also address outstanding comments from the original PR.

Closes #453.

### CODE REVIEW

**Comment by rockbmb** [comment]
Will close #453 .

File: packages/polkadot/src/__snapshots__/polkadot.preimage.e2e.test.ts.snap
[the remark this answers is not in the document]
Comment (andreitrand) [reply packages/polkadot/src/__snapshots__/polkadot.preimage.e2e.test.ts.snap]: Yes, it was. I've removed the snapshot.

File: packages/shared/src/helpers/index.ts [hunk 1]
```diff
@@ -107,6 +107,79 @@ export type BlockProvider = 'Local' | 'NonLocal'
...
+
+  const agenda = calls.map(({ call, origin }) => [
+    [scheduledBlock],
+    [
+      {
+        call,
+        origin: origin,
+      },
+    ],
+  ])
```
[the remark this answers is not in the document]
Comment (andreitrand) [reply packages/shared/src/helpers/index.ts]: Fixed.
